odoo/addons/base/models/res_country.py

```diff
@@ -150,15 +151,50 @@ def _check_address_format(self):
                 except (ValueError, KeyError):
                     raise UserError(_('The layout contains an invalid format key'))
 
+    @api.depends('country_group_ids')
+    def _compute_country_group_codes(self):
+        '''If a country has no associated country groups, assign [''] to country_group_codes.
+        This prevents storing [] as False, which helps avoid iteration over a False value and
+        maintains a valid structure.
+        '''
+        for country in self:
+            country.country_group_codes = country.country_group_ids.mapped('code') or ['']
+
+    def _belongs_to_country_group(self, country_group):
```

Comment:
This function is useless, let's use the same mechanism in the views and in python with `code in self.country_id.country_group_codes` for instance